cli/azd/resources/starter/.github/workflows/azd-up.yml

```diff
@@ -0,0 +1,73 @@
+name: Deploy up to Azure
+
+on:
+  workflow_dispatch:
+  push:
+    # Run when commits are pushed to mainline branch (main or master)
+    # Set this to the mainline branch you use
+    branches:
+      - main
```

Comment:
I think that when we create the federated identity we have to create a trust relationship for each branch we want to support.  Today it looks like we create trust relationships for `main` and pull requests.  We should add `master` to this list given that we are scaffolding this to support both, I think.

Reply:
Do you think we can maybe just include only `main` in this list? The user can update their mainline branch if needed. And `azd pipeline config` needs to be more flexible, so I'd agree with that assessment.

Reply:
Created #2040